src/wallet/wallet.cpp

```diff
@@ -2545,6 +2567,21 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
         }
     }
 
+    if (GetBoolArg("-walletrejectlongchains", DEFAULT_WALLET_REJECT_LONG_CHAINS)) {
+        // Lastly, ensure this tx will pass the mempool's chain limits
```

Comment:
I don't care that it's called only once - this just doesn't seem like code that belongs in the wallet (which shouldn't need to know about the mempool's policy limits as much as possible). Having it abstracted probably makes it usable for other purposes too.

Also, #9290 does not mean we can't do belt and suspender checks to avoid producing long chains. Not adding such a transaction to the wallet (and mempool) is still superior to just adding it to the wallet and relying on retransmit to get it to go through at a later point in time.